authorization/rbac/rbac.go

```diff
@@ -0,0 +1,112 @@
+package rbac
+
+import (
+	"path"
+
+	"github.com/grepory/storage/apis/rbac"
+	"github.com/grepory/storage/authorization"
+	"github.com/grepory/storage/storage"
+)
+
+// Authorizer implements an authorizer interface using Roles Base Acccess
+// Control (RBAC)
+type Authorizer struct {
+	store storage.Store
+}
+
+// Authorize determines if a request is authorized based on its attributes
+func (a *Authorizer) Authorize(attrs authorization.RequestAttributes) (bool, error) {
+	// Get cluster roles binding
+	clusterRoleBindings := []rbac.ClusterRoleBinding{}
+	if err := a.store.List("clusterrolebindings", &clusterRoleBindings); err != nil {
+		return false, err
+	}
+
+	// Inspect each cluster role binding
+	for _, clusterRoleBinding := range clusterRoleBindings {
+		// Verify if this cluster role binding matches our user
+		if !matchesUser(attrs.User, clusterRoleBinding.Subjects) {
+			continue
+		}
+
+		// Get the cluster role that matched our user
+		clusterRole := &rbac.ClusterRole{}
+		key := path.Join("clusterroles", clusterRoleBinding.RoleRef.Name)
```

Comment:
I see now why they use clients everywhere to access objects in durable storage. We should decide if we want to carry tech debt like this forward. We're going to be duplicating this code in the RBAC API.

Reply:
IMO definitely something we should address before moving forward!